### DISCUSSION TITLE
How does the quest fix process work?

### DISCUSSION THREAD

**Discussion by HuangStomach** [TrinityCore/TrinityCore#28990] [Q&A] [answered]
Please let me know if there are any areas where I have not found the correct documentation for the instructions.

I tried to fix the quest [Stand Your Ground](https://www.wowhead.com/quest=59927) and the problem manifests itself in not summoning or converting an NPC. From sniffing I can see that some spells like [Summon Throg - Combat Training (DNT)]( https://www.wowhead.com/spell=325108/summon-throg-combat-training-dnt) are missing. I followed this idea to fix it, found a hook called [QuestScript](https://github.com/TrinityCore/TrinityCore/commit/422ad436c70e3f2f0c67794b3b902d3b398689c0) and tried to implement it. However, when trying to release the spell [Summon Throg - Combat Training (DNT)](https://www.wowhead.com/spell=325108/summon-throg-combat-training-dnt), I found that many of the parameters mentioned in the sniff were missing. It was also difficult to find the exact table in the database where the spell should be located, and no scripts were found in the repository to implement it based on this hook, which left me confused.

I wonder what the exact process is for fixing a quest like this? Or if there is a submission I can draw on.

**Comment by mdX7** [comment, marked as answer]
Sadly thats a not so trivial quest to implement. Its using some not yet fully implemented systems. Related to smooth phasing changes caused by summons. After you accept the quest, the questgiver npc is replaced by a different one, which runs away and you then have to attack. (Spawn, replace should happen caused by summon spell)
When fight is over he returns to his original position and then is replaced by original mob again. The replacing stuff is the stuff which isn't properly implemented yet. Its utilizing ReplaceObject field in SMSG_UPDATE_OBJECT. Also when hes going back to his original position: currently on tc the phase change to make original questgiver visible again happens directly after quest is complete. On retail the phasechange happens when summoned Throg reaches his original position. The phase change causes the replace then.

In general for quest scripting you try to use ScriptedAI to script npcs wherever possible. But you can also use QuestScript, ConversationScript etc, whatever is needed. Another possibility is using SmartAI and implementing related stuff in smart_scripts table. Sometimes SmartAI is a bit too limited, thats when you are forced to use C++ script. 
Scriptingwise you always try to replicate sniff behavior.

I'd also recommend joining our Discord, we are usually faster answering there: https://discord.trinitycore.org/
